### DISCUSSION TITLE
Using 12c to write to a 16 bit register

### DISCUSSION THREAD

**Discussion by boonsiong123** [MrYsLab/telemetrix#74] [Q&A] [answered]
Hi...I am using a 16bit ADC (ADS1115). I am trying to configure the ADC. However, the config register in ADS1115 is 16bit. The args in i2c_write allows up to a value of 255. Any way I can get around with this?

**Comment by Pi3rr3-aprisium** [comment]
Same problem here

**Reply by MrYsLab** [reply, marked as answer]
Thanks for the question. This is accomplished by converting the 16-bit value into two 8-bit values, an MSB and an LSB value. These values are written to the i2c device register.

[This article](https://github.com/MrYsLab/telemetrix/discussions/url) explains how to split a 16-bit value into msb and lsb values using Python.

If you look at line 491 of [Rob Tilaart's Arduino library for the device](https://github.com/RobTillaart/ADS1X15/blob/master/ADS1X15.cpp), you will see an example in an Arduino library.

Here is how this might be expressed in Telemetrix:
```
def write_16_bit_i2c(register, value):
    """
    Write a 16 value to an i2c register
    :param register: i2c register
    :param value: 16 bit value
    :return: None
    """
    msb = value >> 8
    lsb = value & 0xff
    my_board.i2c_write( i2c_device_address, [register,msb, lsb])
```

Please note that this is untested code, since I do not own the device.

Please let me know if this answers your question.

**Reply by Pi3rr3-aprisium** [reply]
Hey, here to update that what you sent is correct and working. We need to write to the board the read from it. thanks again for your help

**Reply by MrYsLab** [reply]
My pleasure to help. I am glad it is working.

**Comment by MrYsLab** [comment]
I just updated my answer since I don't believe it was correct. Again, this is not tested, so if you have any questions, please let me know.

**Reply by boonsiong123** [reply]
Thanks, I will give it a try
